Console/VaultMigrate.php

```diff
@@ -0,0 +1,347 @@
+            ->where('attribute_code = ?', 'braintree_customer_id')
+            ->from(self::EAV_ATTRIBUTE_TABLE, self::ATTRIBUTE_ID);
+        return $db->fetchOne($select);
+    }
+
+    /**
+     * @param $db
+     * @param $eavAttributeId
+     * @return mixed
+     */
+    private function getStoredCards(AdapterInterface $db, OutputInterface $output, $eavAttributeId)
+    {
+        $select = $db->select()
+            ->join('customer_entity', 'customer_entity.entity_id = customer_entity_varchar.entity_id')
+            ->where(self::ATTRIBUTE_ID . ' = ?', $eavAttributeId)
+            ->from(self::CUSTOMER_ENTITY_TABLE, self::VALUE . ' as braintree_id');
+        $result = $db->fetchAll($select);
+
+        if ($result) {
+            $output->writeln('<info>'. count($result) .' stored cards found</info>');
+            return $result;
+        }
+    }
+
+    /**
+     * @param $output
+     * @param $storedCards
+     * @return array
+     */
+    private function findBraintreeCustomers(OutputInterface $output, $storedCards): array
+    {
+        $customers = [];
+        foreach ($storedCards as $storedCard) {
+            $output->writeln('<info>Search Braintree for Customer ID ' . $storedCard['braintree_id'] . '...</info>');
+            $customers[] = $this->braintreeAdapter->getCustomerById($storedCard['braintree_id']);
+        }
+
+        return $customers;
+    }
+
+    /**
+     * @param $customers
+     * @return array
+     */
+    public function remapCustomerData($customers): array
+    {
+        $remappedCustomerData = [];
+
+        foreach ($customers as $customer) {
+            $customerData = [
+                'braintree_id' => $customer->id,
+                'email' => $customer->email
+            ];
+
+            if ($customer->creditCards) {
+                // grab each stored credit card
+                foreach ($customer->creditCards as $creditCard) {
+                    $customerData['storedCards'][] = [
+                        'token' => $creditCard->token,
+                        'expirationMonth' => $creditCard->expirationMonth,
+                        'expirationYear' => $creditCard->expirationYear,
+                        'last4' => $creditCard->last4,
+                        'cardType' => self::CC_MAPPER[str_replace(' ', '-', strtolower($creditCard->cardType))]
+                    ];
+                }
+            }
+
+            // Add customer data to the main customer array
+            $remappedCustomerData[] = $customerData;
+        }
+
+        return $remappedCustomerData;
+    }
+
+    /**
+     * @param array $customers
+     */
+    private function migrateStoredCards(OutputInterface $output, array $customers)
+    {
+        foreach ($customers as $customer) {
+            try {
+                // Customer entity can only have one unique email assigned, so this method is acceptable to use.
+                $m2Customer = $this->customerRepository->get($customer['email']);
```

Comment:
Email is unique per store so we can have 2 or more users with the same email. May need to test if this handles multi-store setups.